File: pkgs/python/wrappers/salesman_euclid_brute_force.py
#!/usr/bin/env python3

import sys
import os

scripts_path = os.path.dirname(os.path.realpath(__file__))

sys.path.insert(0, os.path.realpath(os.path.join(scripts_path, '..')))

from opticontest.problems.salesman_euclid.common import Points
from opticontest.problems.salesman_euclid.brute_force import Algorithm

count = int(input())

points = Points()

for x in range(0, count):
    points.add(*map(float, input().split()))

algorithm = Algorithm(points)

print(' '.join(map(str, algorithm.solution.indices)))

File: pkgs/ruby/wrappers/salesman_euclid_brute_force.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

lib = File.expand_path(File.join('..', 'lib'), __dir__).freeze
$LOAD_PATH.unshift lib unless $LOAD_PATH.include? lib

require 'opticontest/problems/salesman_euclid/common'
require 'opticontest/problems/salesman_euclid/brute_force'

points = Opticontest::Problems::SalesmanEuclid::Common::Points.new

Integer($stdin.gets).times do
  points.add(*$stdin.gets.split.map(&method(:Float)))
end

algorithm =
  Opticontest::Problems::SalesmanEuclid::BruteForce::Algorithm.new points

puts algorithm.solution.indices.join ' '
